src/java.desktop/macosx/classes/com/apple/laf/AquaButtonUI.java

```diff
@@ -307,8 +340,10 @@ public void paint(final Graphics g, final JComponent c) {
         // performs icon and text rect calculations
         final String text;
         final View v = (View)c.getClientProperty(BasicHTML.propertyKey);
-        if (v != null) {
-            // use zero insets for view since layout only handles text calculations
+        if (v != null && c instanceof AbstractButton
+                && ((AbstractButton) c).getIcon() == null) {
```

Comment:
We already know for certain that `c` is an AbstractButton. The first thing in the paint call, the component is casted. Why cast again when we can reuse the variable:
```suggestion
        if (v != null && b.getIcon() == null) {
```

(also on ln 309 we could do the same thing)

Reply:
Totally correct. Updated

Reply:
@SWinxy Looks cleaner. Thanks for the suggestion.